File: app/src/main/java/com/kota/Bahamut/Pages/BoardPage/BoardExtendHistoryItemView.java
package com.kota.Bahamut.Pages.BoardPage;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.kota.Bahamut.DataModels.Bookmark;
import com.kota.Bahamut.R;

public class BoardExtendHistoryItemView extends LinearLayout {
    private View _divider_top = null;
    private TextView _title_label = null;

    public BoardExtendHistoryItemView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public BoardExtendHistoryItemView(Context context) {
        super(context);
        init();
    }

    public void setBookmark(Bookmark bookmark) {
        if (bookmark != null) {
            setTitle(bookmark.getKeyword());
            return;
        }
        clear();
    }

    private void init() {
        ((LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE)).inflate(R.layout.board_extend_optional_page_history_item_view, this);
        _title_label = findViewById(R.id.BoardExtendOptionalPage_historyItemView_Title);
        findViewById(R.id.BoardExtendOptionalPage_historyItemView_Status).setVisibility(GONE);
        findViewById(R.id.BoardExtendOptionalPage_historyItemView_SplitView).setVisibility(GONE);
        findViewById(R.id.BoardExtendOptionalPage_historyItemView_ArrowView).setVisibility(GONE);
        _divider_top = findViewById(R.id.BoardExtendOptionalPage_bookmarkItemView_DividerTop);
    }

    public void setDividerTopVisible(boolean visible) {
        if (_divider_top == null) {
            return;
        }
        if (visible) {
            if (_divider_top.getVisibility() != View.VISIBLE) {
                _divider_top.setVisibility(View.VISIBLE);
            }
        } else if (_divider_top.getVisibility() != View.GONE) {
            _divider_top.setVisibility(View.GONE);
        }
    }

    public void setTitle(String title) {
        if (_title_label != null) {
            if (title == null || title.length() == 0) {
                _title_label.setText("未輸入");
            } else {
                _title_label.setText(title);
            }
        }
    }

    public void clear() {
        setTitle(null);
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/boardPage/BoardExtendHistoryItemView.kt
package com.kota.Bahamut.pages.boardPage

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.kota.Bahamut.dataModels.Bookmark
import com.kota.Bahamut.R

class BoardExtendHistoryItemView : LinearLayout {
    private var dividerTop: View? = null
    private var titleLabel: TextView? = null

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?) : super(context) {
        init()
    }

    fun setBookmark(bookmark: Bookmark?) {
        if (bookmark != null) {
            setTitle(bookmark.keyword)
            return
        }
        clear()
    }

    private fun init() {
        (context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater).inflate(
            R.layout.board_extend_optional_page_history_item_view,
            this
        )
        titleLabel = findViewById(R.id.BoardExtendOptionalPage_historyItemView_Title)
        findViewById<View?>(R.id.BoardExtendOptionalPage_historyItemView_Status)!!.visibility = GONE
        findViewById<View?>(R.id.BoardExtendOptionalPage_historyItemView_SplitView)!!.visibility =
            GONE
        findViewById<View?>(R.id.BoardExtendOptionalPage_historyItemView_ArrowView)!!.visibility =
            GONE
        dividerTop = findViewById(R.id.BoardExtendOptionalPage_bookmarkItemView_DividerTop)
    }

    fun setDividerTopVisible(visible: Boolean) {
        if (dividerTop == null) {
            return
        }
        if (visible) {
            if (dividerTop?.visibility != VISIBLE) {
                dividerTop?.visibility = VISIBLE
            }
        } else if (dividerTop?.visibility != GONE) {
            dividerTop?.visibility = GONE
        }
    }

    fun setTitle(title: String?) {
        if (titleLabel != null) {
            if (title == null || title.isEmpty()) {
                titleLabel?.text = "未輸入"
            } else {
                titleLabel?.text = title
            }
        }
    }

    fun clear() {
        setTitle(null)
    }
}
